pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -123,7 +125,7 @@ func (p *Provider) getStaticParameters(ctx context.Context, instanceType *cloudp
 
 	subnetID := lo.Ternary(nodeClass.Spec.VNETSubnetID != nil, lo.FromPtr(nodeClass.Spec.VNETSubnetID), options.FromContext(ctx).SubnetID)
 
-	if options.FromContext(ctx).NetworkPlugin == consts.NetworkPluginAzure && options.FromContext(ctx).NetworkPluginMode == consts.NetworkPluginModeOverlay {
+	if isAzureCNIOverlay(ctx) {
```

Comment:
this could go into a shared bit of utils, because this isn't the only place we check if we are using azure cni overlay

(operator.go, somewhere in instance.go)